src/utils/hooks/useGetApplicationStructure.tsx

```diff
@@ -32,7 +32,7 @@ const useGetApplicationStructure = ({
   shouldRevalidate = false,
   minRefetchTimestampForRevalidation = 0,
   firstRunValidation = true,
-  shouldCalculateProgress = true,
+  shouldCalculateProgress = structure.info.current.status !== ApplicationStatus.Completed,
```

Comment:
This was the main cause of the problem -- was defaulting to `true` even for completed applications.

(I'm not sure *why* it calculated incomplete progress for completed applications, but I didn't dig into that logic).